From 4d9c6141dd3804a11a52c266445fc778bf7fca05 Mon Sep 17 00:00:00 2001
From: Radostin Stoyanov <rstoyanov1@gmail.com>
Date: Wed, 11 Oct 2017 12:35:57 +0100
Subject: [PATCH 08/45] Convert the output of range() to list

In Python 2 the range() function returns a list [1].
In Python 3 the range function returns a range type [2] which
represents an immutable sequence of numbers [3].

[1] https://docs.python.org/2.7/library/functions.html#range
[2] https://docs.python.org/3/library/functions.html#func-range
[3] https://docs.python.org/3/library/stdtypes.html#typesseq-range
---
 virtManager/details.py     | 2 +-
 virtManager/host.py        | 4 ++--
 virtManager/storagelist.py | 2 +-
 virtinst/util.py           | 2 +-
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/virtManager/details.py b/virtManager/details.py
index affdec38..45fe5625 100644
--- a/virtManager/details.py
+++ b/virtManager/details.py
@@ -3228,7 +3228,7 @@ class vmmDetails(vmmGObjectUI):
         for dev in self.vm.get_panic_devices():
             update_hwlist(HW_LIST_TYPE_PANIC, dev)
 
-        devs = range(len(hw_list_model))
+        devs = list(range(len(hw_list_model)))
         devs.reverse()
         for i in devs:
             _iter = hw_list_model.iter_nth_child(None, i)
diff --git a/virtManager/host.py b/virtManager/host.py
index 2f22642b..1475d7f6 100644
--- a/virtManager/host.py
+++ b/virtManager/host.py
@@ -43,11 +43,11 @@ EDIT_NET_IDS = (
 EDIT_NET_NAME,
 EDIT_NET_AUTOSTART,
 EDIT_NET_QOS,
-) = range(3)
+) = list(range(3))
 
 EDIT_INTERFACE_IDS = (
 EDIT_INTERFACE_STARTMODE,
-) = range(200, 201)
+) = list(range(200, 201))
 
 
 class vmmHost(vmmGObjectUI):
diff --git a/virtManager/storagelist.py b/virtManager/storagelist.py
index a62007f8..78bfad21 100644
--- a/virtManager/storagelist.py
+++ b/virtManager/storagelist.py
@@ -36,7 +36,7 @@ from .createvol import vmmCreateVolume
 EDIT_POOL_IDS = (
 EDIT_POOL_NAME,
 EDIT_POOL_AUTOSTART,
-) = range(2)
+) = list(range(2))
 
 VOL_NUM_COLUMNS = 7
 (VOL_COLUMN_KEY,
diff --git a/virtinst/util.py b/virtinst/util.py
index 2a7bc4d3..e6aa994f 100644
--- a/virtinst/util.py
+++ b/virtinst/util.py
@@ -144,7 +144,7 @@ def generate_name(base, collision_cb, suffix="", lib_collision=True,
         else:
             return collision_cb(tryname)
 
-    numrange = range(start_num, start_num + 100000)
+    numrange = list(range(start_num, start_num + 100000))
     if not force_num:
         numrange = [None] + numrange
 
-- 
2.16.2

